### PR TITLE
[Collections] Auth tokens need to be mutable

### PR DESCRIPTION
**Pull request by yim-lee**
rdar://77095793

### CODE REVIEW

**Comment by yim-lee** [comment]
@swift-ci please smoke test

File: Sources/PackageCollections/PackageCollections.swift [hunk 1]
```diff
@@ -67,6 +67,14 @@ public struct PackageCollections: PackageCollectionsProtocol {
         }
     }
 
+    public mutating func updateAuthTokens(_ authTokens: [AuthTokenType: String]?) {
```
Comment (yim-lee) [inline Sources/PackageCollections/PackageCollections.swift:70]: Not my favorite approach. Open to other options. I don't think the API should be added to `PackageCollectionsProtocol` though.

Comment (tomerd) [reply Sources/PackageCollections/PackageCollections.swift:70]: what about a setter?

Comment (tomerd) [reply Sources/PackageCollections/PackageCollections.swift:70]: also another approach is to define Configuration::authToken to take a function that is called to retrieve the auth values


```
var authTokens = () -> [AuthTokenType: String]?
```

Comment (yim-lee) [reply Sources/PackageCollections/PackageCollections.swift]: > also another approach is to define Configuration::authToken to take a function

I like this. Thanks. Changed impl to this instead. https://github.com/apple/swift-package-manager/pull/3443/commits/bb67334894b5794c307ccf31cb7bb4ada7bcf015

**Comment by yim-lee** [comment]
@swift-ci please smoke test

**Comment by yim-lee** [comment]
@swift-ci please smoke test

**Comment by tomerd** [comment]
@swift-ci please smoke test macOS

**Review by tomerd — APPROVED** [review]

**Review by neonichu — APPROVED** [review]

File: Sources/PackageCollections/PackageCollections.swift (hunk 1 above)
Comment (yim-lee) [reply Sources/PackageCollections/PackageCollections.swift]: The approach in https://github.com/apple/swift-package-manager/pull/3443/commits/bb67334894b5794c307ccf31cb7bb4ada7bcf015 is a breaking change though, so my original approach might be safer... 🤔

Comment (tomerd) [reply Sources/PackageCollections/PackageCollections.swift]: I think its okay to break at this point if we think its the right API, we just need to coordinate landing it cc @neonichu

**Comment by yim-lee** [comment]
Rebased. @swift-ci please smoke test.

**Review by tomerd — APPROVED** [review]

**Pull request merged**